### ISSUE TITLE
meta types (9.0.0) breaks GitHub squashed BREAKING CHANGES footers

### ISSUE THREAD

**Issue by s22chan** [googleapis/release-please#1257]
#### Environment details

  - OS: OSX 12.1
  - Node.js version: v17.2.0
  - npm version: 8.3.0
  - `release-please` version: 13.4.3

#### Steps to reproduce

  1. GitHub squashes multi-commit PRs in the following format:
```
fix: some fix (#1234)

* fix: this becomes the new breaking change comment

* fix: breaking change

BREAKING CHANGE: my comment

* fix: this breaks the footer parser
```

`release-please`@8.2.0 used to parse the first `BREAKING CHANGE:` line to get the CHANGELOG contents to write out; however, starting with 9.0.0, it now grabs `* fix: this becomes the new breaking change comment` which is:

1. not what was intended to warn to users
2. may not be related to the breaking change at all

In the latest 13.4.3 code, the `BREAKING CHANGE:` body node references the entire squashed text as its parent value, which `release-please` extracts until the first newline

**Comment by chingor13** [comment]
Can you post an example with the `BREAKING CHANGE:` section?

Also, it was not intended to parse GitHub's squash/merge format and extra commit message - if you're squash/merging it's likely to avoid having multiple commit messages in history.

As a workaround in v13, there's an annotation you can add to the body of the merged PR that will override the commit message considered for release notes: https://github.com/googleapis/release-please#how-can-i-fix-release-notes

**Comment by s22chan** [comment]
hi, that override won't work because `release-please` no longer tries to find a PR on GraphQL anymore from a squashed commit. 

The example commit above would be:
```markdown
PR title (#PR_Number)
* first commit summary

body

footer

* 2nd commit summary

BREAKING CHANGE: my comment
```
the output from release-please will be:

```markdown
### ⚠ BREAKING CHANGES

* * first commit summary

### Features
...
```

**Comment by chingor13** [comment]
I created an example test for parsing a squash merged commit message in #1265. It looks like it's correctly parsing and generating changelog notes correctly to me.

**Comment by s22chan** [comment]
Sorry for the wrong test case: turns out there's a few more factors at play.  I've tested your branch and made a minimal repro:

```
fix: some fix (#1234)

* fix: this becomes the new breaking change comment

* fix: breaking change

BREAKING CHANGE: my comment

* fix: this breaks the footer parser
```

**Comment by chingor13** [comment]
> ```
> fix: some fix (#1234)
> 
> * fix: this becomes the new breaking change comment
> 
> * fix: breaking change
> 
> BREAKING CHANGE: my comment
> 
> * fix: this breaks the footer parser
> ```

Thanks for this example. Running this through our tests yields something like:

```
### ⚠ BREAKING CHANGES

* * fix: this becomes the new breaking change comment

### Bug Fixes

* some fix ([#1234](https://github.com/googleapis/java-asset/issues/1234)) ([48ea3ab](https://github.com/googleapis/java-asset/commit/48ea3ab7c929c1ed2233dc1fbec95f88))
```

This does look incorrect. @bcoe I'm not entirely sure how that commit is expected to be parsed.
